server/src/middleware/auth.ts

```diff
@@ -0,0 +1,58 @@
+import { NextFunction, Request, Response } from 'express';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { BaseUser, FullUser } from '../../../shared/User';
+import validator from 'validator';
+import { verify } from 'jsonwebtoken';
+
+/**
+ * Basic user authentication for both {@link BaseUser } and {@link FullUser }.
+ *
+ * A JWT must be in the `authorization` header of the request.
+ *
+ * Returns:
+ * - 401 if any checks fail.
+ */
+export function auth(req: Request, res: Response, next: NextFunction): void {
+    try {
+        const token = req.headers.authorization;
```

Comment:
sorry for the comment spam lol- in future this would be of the form `req.cookies.token` if using the cookie parser middleware